ircbot/commands_impl.go

```diff
@@ -69,7 +69,50 @@ func LastPlayed(e Event) error {
 	return nil
 }
 
-func StreamerQueue(e Event) error       { return nil }
+func StreamerQueue(e Event) error       {
+	// Get queue from streamer
+	songQueue, err := e.Bot.Streamer.Queue(e.Context())
+	if err != nil {
+		return err
+	}
+
+	// If the queue is empty then we're done
+	if len(songQueue) == 0 {
+		e.EchoPublic("No queue at the moment")
+		return nil
+	}
+
+	// Calculate playback time for the queue
+	var totalQueueTime time.Duration
+	for _, song := range songQueue {
+		totalQueueTime += song.Length
+	}
+
+	// Define the message strings
+	message := "{green}Queue (/r/ time: %s):{clear} %s"
+	messageJoin := " {red}|{clear} "
+
+	// Grab metadata and color green if requestable
+	onlyMetadata := make([]string, len(songQueue))
+	for i, song := range songQueue {
+		if song.IsUserRequest {
+			onlyMetadata[i] = "{green}" + song.Metadata + "{clear}"
```

Comment:
User supplied data shouldn't be inserted directly into the output message like this, the way our formatting works is two-step; it first does all the IRC color coding (`{green}`, `{clear}`, etc) and then it does the normal formatting (`%s`, `%d`, etc). So if you don't use these a song that contains any of the color codings or even the normal formatting will cause unexpected output. So all user output should be given as argument to the formatting functions, either `Fmt` or one of the `Echo`'s